gdx/src/com/badlogic/gdx/maps/tiled/BaseTmxMapLoader.java

```diff
@@ -634,13 +634,14 @@ protected abstract void addStaticTiles (FileHandle tmxFile, ImageResolver imageR
 
 	protected void addTileProperties (TiledMapTile tile, Element tileElement) {
 		String terrain = tileElement.getAttribute("terrain", null);
-		if (terrain != null) {
-			tile.getProperties().put("terrain", terrain);
-		}
-		String probability = tileElement.getAttribute("probability", null);
-		if (probability != null) {
-			tile.getProperties().put("probability", probability);
-		}
+		tile.setTerrain(terrain);
+
+		float probability = tileElement.getFloatAttribute("probability", 1f);
```

Comment:
apparently default is 0 in tiled spec, see https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#tile
however it might be better to keep it as Float (default null) in order to keep the previous behavior.

Reply:
I've set it to 1 as it defaults to 1.000 in the tiled editor. I however could change it to 0 though.

![image](https://user-images.githubusercontent.com/11274700/118033003-94d6ae00-b368-11eb-8315-9b39b5713dae.png)

Previous behavior was a string, should I change it back to that instead?

Reply:
Indeed, it's maybe more safe to keep it as String. Maybe some user code use it as string and encode some stuff in it, who knows..